rpc/core/events.go

```diff
@@ -111,18 +118,27 @@ func Subscribe(wsCtx rpctypes.WSRPCContext, query string) (*ctypes.ResultSubscri
 			select {
 			case msg := <-sub.Out():
 				resultEvent := &ctypes.ResultEvent{Query: query, Data: msg.Data(), Tags: msg.Tags()}
-				wsCtx.TryWriteRPCResponse(
+				ctx.WSConn.TryWriteRPCResponse(
 					rpctypes.NewRPCSuccessResponse(
-						wsCtx.Codec(),
-						rpctypes.JSONRPCStringID(fmt.Sprintf("%v#event", wsCtx.Request.ID)),
+						ctx.WSConn.Codec(),
+						rpctypes.JSONRPCStringID(fmt.Sprintf("%v#event", ctx.JSONReq.ID)),
 						resultEvent,
 					))
 			case <-sub.Cancelled():
-				wsCtx.TryWriteRPCResponse(
-					rpctypes.RPCServerError(rpctypes.JSONRPCStringID(
-						fmt.Sprintf("%v#event", wsCtx.Request.ID)),
-						fmt.Errorf("subscription was cancelled (reason: %v)", sub.Err()),
-					))
+				if sub.Err() != tmpubsub.ErrUnsubscribed {
+					// should not happen
```

Comment:
the comment `should not happen` does not help understanding what is going on here. Looking at the lines of code below it seems like this can happen, e.g. when tendermint exited, or the subscription was cancelled?

Reply:
> Looking at the lines of code below it seems like this can happen, e.g. when tendermint exited, or the subscription was cancelled?

Yes, but that **should not happen** normally.